bin/oneoff/move_user_storage_ids_to_dashboard.rb

```diff
@@ -0,0 +1,73 @@
+require File.expand_path('../../../pegasus/src/env', __FILE__)
+require src_dir 'database'
+require 'optparse'
+
+# To run: ruby bin/oneoff/move_user_storage_ids_to_dashboard.rb
+# To revert: ruby bin/oneoff/move_user_storage_ids_to_dashboard.rb --revert
+
+# Parse options
+options = {
+  revert: false,
+}
+
+OptionParser.new do |opts|
+  opts.banner = <<~BANNER
+    Usage: #{File.basename(__FILE__)} [options]
+
+    This moves the user_storage_ids table to the Dashboard DB and renames it to user_project_storage_ids
+
+    Options:
+  BANNER
+
+  opts.on('--revert',
+    'Will revert the Database to its previous state (drop the view and move the table back to Pegasus)'
+  ) do |revert|
+    options[:revert] = revert
+  end
+
+  opts.on('-h', '--help', 'Prints this help message') do
+    puts opts
+    exit
+  end
+end.parse!
+puts "Options: #{options}"
+options.freeze
+
+$revert = options[:revert]
+
+def migrate_user_storage_ids_to_dashboard
+  pegasus_user_storage_ids = "#{CDO.pegasus_db_name}__user_storage_ids".to_sym
+  dashboard_user_project_storage_ids = "#{CDO.dashboard_db_name}__user_project_storage_ids".to_sym
+
+  DB.transaction do
```

Comment:
Each schema change (DDL) SQL statement is a separate transaction in MySQL, so I suggest that we do not declare a transaction around the DDL statements.

Reply:
When I wrote this I was thinking that if the table renames fails we wouldn't want to try to make a view, is that something worth being concerned about?